ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/ethernet_interfaces.py

```
@@ -112,6 +112,12 @@ def ethernet_interfaces(self) -> dict | None:
                                         interface["ospf_authentication"] = ospf_authentication
                                         interface["ospf_message_digest_keys"] = ospf_keys
 
+                            if (pim_config := get(l3_interface, "pim.enabled")) and get(vrf, "_l3_multicast_enabled"):
```

No need to assign `pim_config` since there is only one possible value `True`. Just write `True` directly under `sparse_mode`

---

Also instead of `get(vrf, "_l3_multicast_enabled")` use the builtin `vrf.get("_l3_multicast_enabled")`

---

The placement of the raise makes us silently ignore pim.enabled if l3_multicast is not enabled on the VRF/tenant. Is that what we want?

---

Thanks.  I've now split the checks up and they will now raise separate errors for invalid combinations of input.